src/App.js

```diff
@@ -1,24 +1,47 @@
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
+import foodsJson from './foods.json';
+import FoodBox from './components/FoodBox';
+import AddFoodForm from './components/AddFoodForm';
+import { Row } from 'antd';
+import Search from './components/Search';
 
 function App() {
+ const [food, setFoodList] = useState(foodsJson);
+  const [showFood, setShowFood] = useState(foodsJson);
+
+  const addFood = (newFood) => {
+    const updatedFood = [newFood, ...food];
+    setFoodList(updatedFood);
+  };
+
+  const searchFoods = (query) => {
+    const filteredFoods = food.filter((food) =>
+    food.name.toLowerCase().includes(query.toLowerCase())
+    );
+    setShowFood(filteredFoods);
+};
+
+const deleteFood = (foodsName) => {
+  const filteredFoods = food.filter((food) => {
+    return food.name !== foodsName;
```

Comment:
Your `deleteFood()` function is not working properly because you're only updating `showFood` with `setShowFood` you need to update `food` as well.
```
  const deleteFood = (foodsName) => {
    const filteredFoods = food.filter((food) => {
      return food.name !== foodsName });
    setShowFood(filteredFoods);
    setFoodList(filteredFoods);
  };
```